kratos/utilities/geometrical_projection_utilities.h

```diff
@@ -199,6 +198,35 @@ class GeometricalProjectionUtilities
         return norm_2(rPointProjected.Coordinates()-p_c);
     }
 
+
+    /**
+     * @brief Computes the minimal distance to a line
+     * @details Projects over a line and if the point projected is inside the line that distance is taken into consideration, otherwise the minimal between the two points in the line is considered
+     * @tparam TGeometryType The type of the line
+     * @param rGeometry The line where compute the distance
+     * @param rPoint The point to compute the distance
+     * @param Tolerance Tolerance to check it falls inside the line
+     * @return Distance The distance between point and line
+     */
+    template<class TGeometryType>
+    static inline double FastMinimalDistanceOnLine(
+        const TGeometryType& rGeometry,
+        const PointType& rPoint,
+        const double Tolerance = 1.0e-9
+        )
+    {
+        PointType projected_point;
+        const double projected_distance = FastProjectOnLine(rGeometry, rPoint, projected_point);
+        typename TGeometryType::CoordinatesArrayType projected_local;
+        if (rGeometry.IsInside(projected_point.Coordinates(), projected_local, Tolerance)) {
+            return projected_distance;
+        } else {
+            const double distance_a = norm_2(rPoint.Coordinates() - rGeometry[0].Coordinates()); /// TODO: Replace with distance method if ever exists
+            const double distance_b = norm_2(rPoint.Coordinates() - rGeometry[1].Coordinates()); /// TODO: Replace with distance method if ever exists
+            return std::min(distance_a, distance_b);
```

Comment:
Can you please add a version that doesn't take the square root? I imagine there are a lot of cases in which you just need sorting (closest point, farthest point, ...) but not the actual distance. We'd be saving a lot of unnecessary work in those cases. This also applies to the other PR.

This version (the one that computes the distance) could just call the one that computes the square, and take its root so there'd be no code duplication (+ there'd be no performance overhead because this can surely be inlined).
```cpp
double FastMinimalSquaredDistanceOnLine(...)
{...}

double FastMinimalDistanceOnLine(...)
{
    return std::sqrt(FastMinimalSquaredDistanceOnLine(...));
}
```

Reply:
Actually, this comes up so often that I'd just define a dedicated ```Distance``` class that stores the squared distance, and produces its square root on demand only. I think it'd be a good idea to return such an object from every distance calculation.
```cpp
template <class TValue>
class Distance
{
public:
    Norm(TValue squared) : mSquared(squared) {}

    ...

    bool operator<(Norm Left, Norm Right) {...}

    explicit operator TValue() const noexcept {return std::sqrt(mSquared);}

private:
    TValue mSquared;
};
```

*Having an additional ```bool``` member would also allow storing the root if it's already been computed once, which would help to avoid computing the root repeatedly.*

Reply:
See https://github.com/KratosMultiphysics/Kratos/pull/10404

Reply:
> See #10404

I'm talking about avoiding the square root calculation unless it's absolutely necessary. Does that PR address that?

Reply:
Well, can be adapted

Reply:
Done @matekelemen